### PR TITLE
feat: Check for rust version in Makefile

### PR DESCRIPTION
**Pull request by derrix060**
- Enhanced MSRV check in GitHub workflow to include Makefile updates.
- Added a new target in Makefile to ensure Rust version compliance.
- Updated dependencies and formatting targets to include Rust version check.
- Fixed minor formatting issues in Makefile comments.

### CODE REVIEW

**Comment by wojciechos** [comment]
Looks like pipeline [started to fail](https://github.com/NethermindEth/juno/actions/runs/11287790309/job/31394327428?pr=2206#step:3:907) because of `error: 'cargo msrv' requires a subcommand but one was not provided`. Shouldn't it be `cargo msrv` with find subcommand [here](https://github.com/derrix060/detect-rust-minimum-version/blob/main/action.yml#L62)?

**Review by wojciechos — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -91,7 +98,7 @@ lint: install-golangci-lint
 tidy: ## add missing and remove unused modules
 	 go mod tidy
 
-format: ## run go & rust formatters
+format: check-rust ## run go & rust formatters
```
Comment (kirugan) [inline Makefile:101]: it's not critical for format target, maybe remove `check-rust` from here?

Comment (derrix060) [reply Makefile:101]: doesn't it need to have rust to format?  From what I saw it formats both go and rust

Comment (kirugan) [reply Makefile:101]: it does (but in theory having `rustfmt` doesn't mean that you need `rustc`). But my point is that we don't need specific rust version in order to format rust codebase.

Rust from version to version doesn't change language significantly, they use so-called editions to introduce breaking changes into the language. And editions are controlled by cargo.toml

**Review by kirugan — APPROVED** [review]

**Pull request merged**